scrapy/resolver.py

```diff
@@ -4,17 +4,25 @@
 from scrapy.utils.datatypes import LocalCache
 
 # TODO: cache misses
-# TODO: make cache size a setting
 
 dnscache = LocalCache(10000)
 
 class CachingThreadedResolver(ThreadedResolver):
+    def __init__(self, reactor, settings):
```

Comment:
I think there is a value in decoupling the code. E.g. it'd be easier to move it to a module which all Twisted users can use, not only Scrapy users. Inconvenience doesn't look too severe - just some constructor arguments :)